apps/aecore/src/aec_conductor.erl

```diff
@@ -1132,11 +1204,13 @@ handle_add_block(Block, Hash, Prev, #state{top_block_hash = TopBlockHash} = Stat
 handle_successfully_added_block(Block, Hash, false, _, Events, State, Origin) ->
     maybe_publish_tx_events(Events, Hash, Origin),
     maybe_publish_block(Origin, Block),
-    {ok, State};
+    State1 = maybe_consensus_change(State, Block),
+    {ok, State1};
 handle_successfully_added_block(Block, Hash, true, PrevKeyHeader, Events, State, Origin) ->
     maybe_publish_tx_events(Events, Hash, Origin),
     maybe_publish_block(Origin, Block),
-    case preempt_on_new_top(State, Block, Hash, Origin) of
+    State1 = maybe_consensus_change(State, Block),
+    case preempt_on_new_top(State1, Block, Hash, Origin) of
         {micro_changed, State2 = #state{ consensus = Cons }} ->
```

Comment:
Is this correct (creating an entirely new state rather than building on `State1`)?

Reply:
This is a pattern not a expression